### PR TITLE
[15.0][ADD] sale_custom_rounding: Custom taxes rounding method in sale orders

### PR DESCRIPTION
**Pull request by manuelregidor**
This module allows to select a custom taxes rounding method in sale orders. The rounding method can be configured in contacts so it is autmatically selected in a sale order when this contact is set as the invoice address.

### CODE REVIEW

**Comment by manuelregidor** [comment]
@HaraldPanten @ValentinVinagre

**Comment by ValentinVinagre** [comment]
@Jesarregui could u review please?

**Review by Jesarregui — APPROVED** [review]
LGTM

**Comment by ValentinVinagre** [comment]
@etobella la otra parte 😄

File: sale_custom_rounding/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                and sel.partner_invoice_id.tax_calculation_rounding_method
+            ):
+                tax_calculation_rounding_method = (
+                    sel.partner_invoice_id.tax_calculation_rounding_method
+                )
+            sel.tax_calculation_rounding_method = tax_calculation_rounding_method
+
+    @api.depends("tax_calculation_rounding_method")
+    def _compute_tax_totals_json(self):
+        for sel in self.filtered(lambda a: a.tax_calculation_rounding_method):
```
Comment (etobella) [inline sale_custom_rounding/models/sale_order.py]: call it record, not sel, it is confusing

Comment (etobella) [reply sale_custom_rounding/models/sale_order.py]: Also, I think you can simplify the code

you should probably do:

```
super(
            SaleOrder, self.filtered(lambda a: not a.tax_calculation_rounding_method)
        )._compute_tax_totals_json()
super(
            SaleOrder, self.filtered(lambda a: a.tax_calculation_rounding_method == "round_per_line").with_context
            (round=True
        )._compute_tax_totals_json()
super(
            SaleOrder, self.filtered(lambda a: a.tax_calculation_rounding_method == "round_globally").with_context
            (round=False
        )._compute_tax_totals_json()

```

Comment (manuelregidor) [reply sale_custom_rounding/models/sale_order.py]: Done! Thank you.

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by ValentinVinagre** [comment]
@HaraldPanten

File: sale_custom_rounding/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends("order_id.tax_calculation_rounding_method")
+    def _compute_amount(self):
+        for line in self.filtered(lambda a: a.order_id.tax_calculation_rounding_method):
+            super(
+                SaleOrderLine,
+                line.with_context(
+                    round=line.order_id.tax_calculation_rounding_method
```
Comment (etobella) [inline sale_custom_rounding/models/sale_order.py]: Why don't you do something like we did with _compute_tax_totals_json?

Comment (manuelregidor) [reply sale_custom_rounding/models/sale_order.py]: Done, thank you.

**Review by etobella — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**